[ADD] website: page on website elements (#12570)
odoo/documentation: content/applications/websites/website/web_design/elements.rst

========
Elements
========

Elements help structure and present content effectively, they range from text-based components like
:ref:`titles <website/web_design/elements/titles>` and highlighted text to interactive ones such as
:ref:`buttons <website/web_design/elements/buttons>` and :ref:`links
<website/web_design/elements/links>`. Visual elements like :ref:`images
<website/web_design/elements/images>`, :ref:`icons <website/web_design/elements/icons>`,
:ref:`videos <website/web_design/elements/videos>`, and :ref:`animations
<website/web_design/elements/animations>` can be added to represent content better and organize
information.

To modify all elements on website:

#. Click on :guilabel:`edit`.
#. Select the section where you want to add an element.
#. Once changes are done, click on :guilabel:`Save`.

> 

Reviewer: I think I would add a note somewhere around here to mention the fact that:
The default styles for headings, buttons,  links, and paragraph text are defined in the :doc:`Theme tab <add a doc link>` of the website editor.

:guilabel:`Loop`.

.. _website/web_design/elements/icons:

Icons
=====

Icons are visual elements that enhance website design and usability, they help convey information
quickly and improve navigation. To insert an icon type `/image` and open the :guilabel:`Icons` tab.
To modify an icon, click on it and on the website editor, customize it with options like
:guilabel:`Color`, :guilabel:`Size`, :ref:`Animations <website/web_design/elements/animations>`
:guilabel:`Shape`.

.. _website/web_design/elements/links:

Links
=====

Links are essential elements that connect different pages and resources, guiding visitors and
improving navigation. To add a link type `/link`, a pop-up opens where :guilabel:`Label`,
:guilabel:`URL or Email` and :guilabel:`Style` can be modified.

.. _website/web_design/elements/lists:

Lists
=====

Lists help organize content clearly, making information easier to read and improving web pages'
structures. Type `/list`, three different types of lists are available:
:guilabel:`Bulleted lists`, :guilabel:`Numbered lists`, and :guilabel:`Checklists`.

.. _website/web_design/elements/animations:

Animations
==========

> Animations add movement to website elements, enhancing visual aspects and improving visitor
> engagement. There are two different types of animations: :guilabel:`on scroll`,
> :guilabel:`on appearance`. To add an animation:

Reviewer: The second part of the first sentence sounds a bit too marketing-y; I'd remove it.

```suggestion
Animations are used to add movement to website elements, such as images, text, building blocks, etc. Three types of animation are available: :guilabel:`On Scroll`, :guilabel:`On Appearance`, and :guilabel:`On Hover` (for images only). To add an animation:
```

#. Click on an element.

> #. On the website editor click on the :guilabel:`Animation` field.
> #. Choose the type of animation.

Reviewer: ```suggestion
#. In the website editor, go to the relevant section for the website element you wish to animate (e.g., :guilabel:`Button`, :guilabel:`Column`, :guilabel:`Inline Text`, etc.).
#. In the :guilabel:`Animation` field, select the type of animation.
```
I think it's important here to mention where people should go (especially since you can have multiple Animation fields in the website editor depending on what you can animate.

Animations on scroll
--------------------

Depending on the animation, different options are available. For :guilabel:`animations on scroll`
it is possible to:

>  - Choose :guilabel:`in` to add the animation when the element enters the screen
>    and :guilabel:`out` to add it when it leaves the screen.

Reviewer: ```suggestion
 - Choose :guilabel:`In` to add the animation when the element enters the screen
   and :guilabel:`Out` to add it when it leaves the screen.
```
nitpick

 - Choose among the different effects.
 - Choose the :guilabel:`Direction` of different effects.
 - Adapt the :guilabel:`Intensity` of the effect.

>  - Define the :guilabel:`Scroll Zone`, where the first value represents the % of the screen shown

Reviewer: ```suggestion
 - Define the :guilabel:`Scroll Zone`, where the first value represents the percentage of the screen shown
```

   when the effect starts and the second value represents its percentage at the end.

Animations on appearance
------------------------

For :guilabel:`animations on appearance` it is possible to:

 - Choose among different effects.
 - Choose the :guilabel:`Direction` of the effects.
 - Pick a :guilabel:`Trigger` option to have the animation happen the :guilabel:`First Time only`
   or :guilabel:`Every Time`.
 - Adapt the :guilabel:`Intensity` of the effect.
 - Define a :guilabel:`Start After` if you want the animation to be triggered after a number of
   seconds.
 - Choose :guilabel:`Duration` for the animation.

> Animations on images

Reviewer: ```suggestion
Animations on hover (for images only)
```
